metsim/metsim.py

```diff
@@ -326,31 +401,37 @@ def setup_output(self, prototype: xr.Dataset=None):
     def _aggregate_state(self):
         """Aggregate data out of the state file and load it into `met_data`"""
         # Precipitation record
-        begin_record = self.params['start'] - pd.Timedelta("90 days")
-        end_record = self.params['start'] - pd.Timedelta("1 days")
-        record_dates = date_range(begin_record, end_record,
+
+        assert self.state.dims['time'] == 90
```

Comment:
I really don't like this check, because it seems to limit some cases that are pretty useful in my mind.  For example, if you have a long timeseries broken up into 1 year chunks you could previously just use the previous year's data as your state file.  Now it looks like you'd have to preprocess all of those files and create an additional state file for each years input.